tests/workflow-validator.spec.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { ValidationError, WorkflowValidator } from '../src/';
+import { Workflow } from '../src/lib/definitions/workflow';
+
+describe('workflow-validator', () => {
+  it('should return errors instance of ValidationError if the workflow provided is not valid', () => {
+    // @ts-ignore
+    const workflow = {
+      id: 'helloworld',
+      name: 'Hello World Workflow',
+      version: '1.0',
+      description: 'Inject Hello World',
+      start: 'Hello State',
+      states: [],
+    } as Workflow;
+
+    const workflowValidator = new WorkflowValidator(workflow);
+    expect(workflowValidator.errors.length).toBe(0);
+
+    workflowValidator.validate();
```

Comment:
We are forcing here the client to invoke validate() to get the list of errors, 

I am thinking if would be more convenient to have the invocation in the WorkflowValidator constructor (for example) and have an inmutable object.

So from the client point of view it would be: 

```
    expect(new WorkflowValidator(workflow).errors.length).toBe(0);

```

instead of having to invoke validate() before

Reply:
I agree, the validation can be done in the constructor by relying on [utils.validate](https://github.com/serverlessworkflow/sdk-typescript/blob/main/src/lib/utils.ts#L10).